### ISSUE TITLE
App works with localhost or loopback IP but not with device IP

### ISSUE THREAD

**Issue by iamsubingyawali** [stancl/tenancy#837]
### Bug description

I have my Laravel 9 installation with tenancy inside xampp/htdocs. It worked fine when I used API URL with localhost or loopback IP, but it threw 404 when I used my device's IP address.

`http://localhost/myapp/public/api/users` It worked fine and returned me all the users

but

`http://192.168.254.6/myapp/public/api/users` gave me 404

I simply tested the case with only one php file test.php and `http://192.168.254.6/testapp/test.php` worked fine.

I figured out that this error is due to my tenancy configuration. In the `config/tenancy.php` file there is list of central domains where it has been stated in the comment that those central domains are only relevant if I am using domain or subdomain to identify tenants. But I am using request data to identify tenant and it is still preventing me from accessing the app from other URLs except those listed there.

Old config

```
/**
 * The list of domains hosting your central app.
 *
 * Only relevant if you're using the domain or subdomain identification middleware.
 */
'central_domains' => [
    '127.0.0.1',
    'localhost',
],
```

**Fix**

I simply added my local device IP to that list of central domains, and it worked. Also, I found that I can use an empty string to allow all domains. 

New config

```
/**
 * The list of domains hosting your central app.
 *
 * Only relevant if you're using the domain or subdomain identification middleware.
 */
'central_domains' => [
    '127.0.0.1',
    'localhost',
    '',
],
```

Is this the expected behaviour?

### Steps to reproduce

Simply browse the tenancy enabled site with local device IP instead of localhost or loopback IP.

### Expected behavior

The site should open no matter which URL I used among localhost, loopback, and my local network IP.

### Laravel version

9.6.0

### stancl/tenancy version

3.5.5

**Comment by stancl** [comment]
If you're using an IP address, you can't be using domain identification.

And if you're not using domain identification, the `central_domains` config has no effect.

**Comment by iamsubingyawali** [comment]
Exactly. But what I said is a bit different. I am not using the IP address for identification. I am using request data for tenant identification.

The issue has nothing to do with tenant identification instead I am not being able to access the application using my local IP address. I can use localhost and 127.0.0.1 but not my local IP. But after adding my local IP or a blank string to the list of central domains I was able to access.

In short, `central_domains` is preventing me from accessing the app using other URLs than listed there, even when not using domain identification.

**Comment by stancl** [comment]
I'm not sure why the config would have an effect here then? Are you using the `PreventAccessFromCentralDomains` middleware or something like that?

It doesn't seem right that `central_domains` would do anything with request data identification.

**Comment by iamsubingyawali** [comment]
No, I am not using any other middleware than `InitializeTenancyByRequestData` and `CheckTenantForMaintenanceMode`. You can test this behavior by putting a tenant enabled app inside server root and trying to access with local IP.

Surprisingly, the requests don't even show up in the middleware. I created a Logger middleware to check incoming requests and logged the requests inside `public/index.php` and the `Logger` middleware, the requests were reaching to `index.php` but not to the middleware when using IP address. The app responds with 404 before reaching to middlewares.

I could see the request in the middleware while accessing via `http://localhost/myapp/public/api/users` and `http://127.0.0.1/myapp/public/api/users` but not via `http://192.168.254.6/myapp/public/api/users` until I added `192.168.254.6` into the `central_domains`.

**Comment by stancl** [comment]
That's strange. I looked into the code and `central_domains` doesn't appear to be used in anything related to request data identification *but* for some reason the request data test also overrides `central_domains`.

I'll keep this issue closed right now since it isn't technically a bug in any described behavior, but if you could debug this further and see what's causing this, I'm happy to reopen and fix it.

Thanks!

**Comment by iamsubingyawali** [comment]
I think we shouldn't map routes in `RouteServiceProvider` if we are not using domain or subdomain identifiers. In the [documentation](https://tenancyforlaravel.com/docs/v3/quickstart#central-routes), there is instruction to map routes where we are specifically using central domains while mapping.

I think we can leave `RouteServiceProvider` as it is if we are using request data identifier. Am I right?

**Comment by stancl** [comment]
I think so, yeah.

**Comment by iamsubingyawali** [comment]
Thanks. I request you to confirm this and update the docs so that it is clear for everyone.
